src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,180 @@
+            throw new DataProcessingException("Can't get all data from DB", exception);
+        }
+        return carList;
     }
 
     @Override
     public Car update(Car car) {
-        List<Car> dbCar = Storage.getCars();
-        IntStream.range(0, dbCar.size())
-                .filter(i -> dbCar.get(i).equals(car))
-                .findFirst()
-                .ifPresent(i -> dbCar.set(i, car));
+        String queryForDeleteInCarsDrivers = "delete from cars_drivers where car_id = ?;";
+        String queryForUpdateInCars = "update cars set car_deleted = true where car_id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteStmtCarsDrivers = connection
+                        .prepareStatement(queryForDeleteInCarsDrivers);
+                PreparedStatement updateStmtCars = connection
+                        .prepareStatement(queryForUpdateInCars)) {
+            deleteStmtCarsDrivers.setLong(1, car.getId());
+            updateStmtCars.setLong(1, car.getId());
+            deleteStmtCarsDrivers.executeUpdate();
+            updateStmtCars.executeUpdate();
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't update the data - "
+                    + car + "in table of cars or cars_drivers", exception);
+        }
+        create(car);
+        createCarsDriversDB(car);
         return car;
     }
 
     @Override
     public boolean delete(Long id) {
-        return Storage.getCars().removeIf(i -> i.getId().equals(id));
+        String queryForCars = "UPDATE cars SET car_deleted = true where car_id = ?;";
+        String queryForCarsDrivers = "delete from cars_drivers where car_id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statementForCars = connection.prepareStatement(queryForCars);
+                PreparedStatement statementForCarsDrivers = connection
+                        .prepareStatement(queryForCarsDrivers)) {
+            statementForCars.setLong(1, id);
+            statementForCarsDrivers.setLong(1, id);
+            return statementForCars.executeUpdate() > 0
+                    && statementForCarsDrivers.executeUpdate() > 0;
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't delete the data by id: " + id, exception);
+        }
     }
 
     public List<Car> getAllByDriver(Long driverId) {
-        return getAll().stream()
-                .filter(car -> car.getDrivers()
-                        .stream()
-                        .anyMatch(driver -> driver.getId().equals(driverId)))
-                .collect(Collectors.toList());
+        List<Car> carList = new ArrayList<>();
+        String query = "select distinct(c.car_id) from cars_drivers cd"
+                + " inner join cars c on cd.car_id = c.car_id "
+                + " where cd.driver_id = ? and c.car_deleted = false;";
```

Comment:
let's also fetch the info about the manufacturer
and join `drivers` column as well so that we can check that driver is not deleted